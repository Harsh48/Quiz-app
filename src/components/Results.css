.results-container {
    text-align: center;
    background: #fff;
    padding: 20px;
    border-radius: 8px;
    box-shadow: 0 0 10px rgba(0, 0, 0, 0.1);
  }
  
  .results-title {
    font-size: 1.8em;
    margin-bottom: 20px;
  }
  
  .reset-button {
    background-color: #f44336;
    color: white;
    padding: 15px 32px;
    text-align: center;
    font-size: 16px;
    margin: 4px 2px;
    border: none;
    border-radius: 4px;
    cursor: pointer;
  }
  
  .reset-button:hover {
    background-color: #da190b;
  }
  