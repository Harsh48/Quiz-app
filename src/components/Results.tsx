import React from 'react';
import { useSelector } from 'react-redux';
import { RootState } from '../redux/store';
import styled from 'styled-components';

interface AnswerItemProps {
  correct: boolean;
}

const ResultsContainer = styled.div`
  text-align: center;
  background: #fff;
  padding: 20px;
  border-radius: 8px;
  box-shadow: 0 0 10px rgba(0, 0, 0, 0.1);
  animation: fadeIn 1s ease-in;
`;

const ResultsTitle = styled.h2`
  font-size: 1.8em;
  margin-bottom: 20px;
`;

const ResetButton = styled.button`
  background-color: #f44336;
  color: white;
  padding: 15px 32px;
  font-size: 16px;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  transition: background-color 0.3s ease;

  &:hover {
    background-color: #da190b;
  }
`;

const AnswerList = styled.ul`
  list-style: none;
  padding: 0;
`;

const AnswerItem = styled.li<AnswerItemProps>`
  margin-bottom: 10px;
  color: ${(props) => (props.correct ? 'green' : 'red')};
`;

interface ResultsProps {
  onReset: () => void;
}

const Results: React.FC<ResultsProps> = ({ onReset }) => {
  const score = useSelector((state: RootState) => state.quiz.score);
  const answers = useSelector((state: RootState) => state.quiz.answers);

  return (
    <ResultsContainer>
      <ResultsTitle>Your Score: {score}</ResultsTitle>
      <AnswerList>
        {answers.map((answer, index) => (
          <AnswerItem key={index} correct={answer.correct}>
            {answer.question} - Your answer: {answer.answer} ({answer.correct ? 'Correct' : 'Incorrect'})
          </AnswerItem>
        ))}
      </AnswerList>
      <ResetButton onClick={onReset}>Start New Quiz</ResetButton>
    </ResultsContainer>
  );
};

export default Results;
