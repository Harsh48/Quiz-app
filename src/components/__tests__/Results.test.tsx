import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Results from '../Results';
import { RootState } from '../../redux/store';
import 'jest-styled-components';

const mockStore = configureStore([]);

const initialState: RootState = {
  quiz: {
    questions: [],
    currentQuestionIndex: 0,
    score: 30,
    status: 'finished',
    answers: [
      { question: 'What is the capital of France?', answer: 'Paris', correct: true },
      { question: 'What is 2+2?', answer: '4', correct: true },
      { question: 'Is the sky blue?', answer: 'Yes', correct: false },
    ],
  },
};

describe('Results Component', () => {
  test('renders Results component correctly', () => {
    const store = mockStore(initialState);
    const { asFragment } = render(
      <Provider store={store}>
        <Results onReset={jest.fn()} />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('displays score and answers correctly', () => {
    const store = mockStore(initialState);
    render(
      <Provider store={store}>
        <Results onReset={jest.fn()} />
      </Provider>
    );

    expect(screen.getByText('Your Score: 30')).toBeInTheDocument();
    expect(screen.getByText('What is the capital of France? - Your answer: Paris (Correct)')).toBeInTheDocument();
    expect(screen.getByText('What is 2+2? - Your answer: 4 (Correct)')).toBeInTheDocument();
    expect(screen.getByText('Is the sky blue? - Your answer: Yes (Incorrect)')).toBeInTheDocument();
  });

  test('calls onReset when reset button is clicked', () => {
    const store = mockStore(initialState);
    const onReset = jest.fn();

    render(
      <Provider store={store}>
        <Results onReset={onReset} />
      </Provider>
    );

    fireEvent.click(screen.getByText('Start New Quiz'));
    expect(onReset).toHaveBeenCalled();
  });
});
