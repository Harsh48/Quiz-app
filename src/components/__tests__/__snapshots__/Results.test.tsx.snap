// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results Component renders Results component correctly 1`] = `
<DocumentFragment>
  .c0 {
  text-align: center;
  background: #fff;
  padding: 20px;
  border-radius: 8px;
  box-shadow: 0 0 10px rgba(0, 0, 0, 0.1);
  animation: fadeIn 1s ease-in;
}

.c1 {
  font-size: 1.8em;
  margin-bottom: 20px;
}

.c5 {
  background-color: #f44336;
  color: white;
  padding: 15px 32px;
  font-size: 16px;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  transition: background-color 0.3s ease;
}

.c5:hover {
  background-color: #da190b;
}

.c2 {
  list-style: none;
  padding: 0;
}

.c3 {
  margin-bottom: 10px;
  color: green;
}

.c4 {
  margin-bottom: 10px;
  color: red;
}

<div
    class="c0"
  >
    <h2
      class="c1"
    >
      Your Score: 30
    </h2>
    <ul
      class="c2"
    >
      <li
        class="c3"
      >
        What is the capital of France? - Your answer: Paris (Correct)
      </li>
      <li
        class="c3"
      >
        What is 2+2? - Your answer: 4 (Correct)
      </li>
      <li
        class="c4"
      >
        Is the sky blue? - Your answer: Yes (Incorrect)
      </li>
    </ul>
    <button
      class="c5"
    >
      Start New Quiz
    </button>
  </div>
</DocumentFragment>
`;

exports[`renders Results component correctly 1`] = `
<DocumentFragment>
  .c0 {
  text-align: center;
  background: #fff;
  padding: 20px;
  border-radius: 8px;
  box-shadow: 0 0 10px rgba(0, 0, 0, 0.1);
  animation: fadeIn 1s ease-in;
}

.c1 {
  font-size: 1.8em;
  margin-bottom: 20px;
}

.c5 {
  background-color: #f44336;
  color: white;
  padding: 15px 32px;
  font-size: 16px;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  transition: background-color 0.3s ease;
}

.c5:hover {
  background-color: #da190b;
}

.c2 {
  list-style: none;
  padding: 0;
}

.c3 {
  margin-bottom: 10px;
  color: green;
}

.c4 {
  margin-bottom: 10px;
  color: red;
}

<div
    class="c0"
  >
    <h2
      class="c1"
    >
      Your Score: 30
    </h2>
    <ul
      class="c2"
    >
      <li
        class="c3"
      >
        What is the capital of France? - Your answer: Paris (Correct)
      </li>
      <li
        class="c3"
      >
        What is 2+2? - Your answer: 4 (Correct)
      </li>
      <li
        class="c4"
      >
        Is the sky blue? - Your answer: Yes (Incorrect)
      </li>
    </ul>
    <button
      class="c5"
    >
      Start New Quiz
    </button>
  </div>
</DocumentFragment>
`;
